frontend/src/store/UserAccessRoleState.ts

```diff
@@ -0,0 +1,14 @@
+import type { FamApplicationUserRoleAssignmentGet } from 'fam-app-acsctl-api';
+import { shallowRef } from 'vue';
+
+export const userRoleAssignments = shallowRef<
+    FamApplicationUserRoleAssignmentGet[]
+>([]);
+
+// --- Setter
+
+export const setUserRoleAssignments = (
```

Comment:
Hi Ian, I feel normally we set state variables when we want to share the value between different components. But in our case, only the ManagePermission component needs this value, is that better if we pass it as prop as well from router? and still use local variable to handle?

Reply:
Thanks Catherine, initially I feel `userRoleAssignments` should be a global state, due to the fact that "**ManagePermission**" seems to be our top level component (dashboard) already. I may have overlook that time.
 
When I look again, the "**userRoleAssignments**" are shared between "**ManagePermission**" and child component "**UserDataTable**" component.

I wasn't aware this "**userRoleAssignments**" was passed to "**UserDataTable**" component as a **props**. Now I feel even "UserDataTable" does not need this props, it should reflect right away what's changed in the state to show the table. 

Question, do you think in the future "**UserDataTable**" is going to be **used by different tabs**, so the props is needed (I kind of think it is going to be stay only specific for showing and sharing "**userRoleAssignment**") ?

Reply:
Hi @MCatherine1994 I have switched this to be as props. Hope I didn't miss anything.